import java.util.Arrays;

public class stableselectionsort {
  public static void sort(int arr[]) {
    for (int i = 0; i < arr.length - 1; i++) {
      int min = i;
      for (int j = i + 1; j < arr.length; j++) {
        if (arr[j] < arr[min]) {
          min = j;
        }
      }
      int tmp = arr[min];
      int pos = min;
      while (pos > i) {
        arr[pos] = arr[pos - 1];
        pos--;
      }
      arr[pos] = tmp;
    }
  }

  public static void main(String[] args) {
    int[] array = new int[]{0, 39, 21, 62, 91, 77, 14, 23, 90, 69, 51, 81, 68, 83, 32, 56};
    sort(array);
    System.out.println(Arrays.toString(array));
  }
}
